### PR TITLE
8309978: [x64] Fix useless padding

### PR DESCRIPTION
**Pull request by vnkozlov**
Fixed typo in `IntelJccErratum::compute_padding()`.

Due to the typo (`mach` instead of `next`) useless padding could be generated because size of `mach` instruction (which is `cmp` in this case and big)  counted twice. As result combined size most likely cross 32-bytes cache line boundary and padding is generated to avoid that.

```
030    B2: # out( B4 B3 ) <- in( B1 ) Freq: 0.999999 
       nop # 16 bytes pad for loops and calls 
040    cmpb [R12 + R10 << 3 + #144] (compressed oop addressing), #42 
049    jle,s B4 P=0.667944 C=6785.000000
```
Note: only some x86 CPUs are [affected](https://github.com/openjdk/jdk/blob/ba837b4bfa2dea85653d8a8fccd0817a569b4378/src/hotspot/cpu/x86/vm_version_x86.cpp#L1957).

For new IR test to work I moved `PHASE_FINAL_CODE` IR print inside `PhaseOutput` scope because padding nodes (`NOP` mach nodes) are present only in this phase IR.

Added new IR test. Tested tier1-3, xcomp, stress.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8309978](https://bugs.openjdk.org/browse/JDK-8309978): [x64] Fix useless padding (**Enhancement** - P4)

### Reviewers
 * [Christian Hagedorn](https://openjdk.org/census#chagedorn) (@chhagedorn - **Reviewer**)
 * [Tobias Hartmann](https://openjdk.org/census#thartmann) (@TobiHartmann - **Reviewer**) ⚠️ Review applies to [5c711eff](https://git.openjdk.org/jdk/pull/14461/files/5c711efff5667ad46f637511ee460aa26f3fad94)
 * [Aleksey Shipilev](https://openjdk.org/census#shade) (@shipilev - **Reviewer**) ⚠️ Review applies to [5c711eff](https://git.openjdk.org/jdk/pull/14461/files/5c711efff5667ad46f637511ee460aa26f3fad94)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/14461/head:pull/14461` \
`$ git checkout pull/14461`

Update a local copy of the PR: \
`$ git checkout pull/14461` \
`$ git pull https://git.openjdk.org/jdk.git pull/14461/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 14461`

View PR using the GUI difftool: \
`$ git pr show -t 14461`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/14461.diff">https://git.openjdk.org/jdk/pull/14461.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/14461#issuecomment-1590462969)

### CODE REVIEW

File: test/hotspot/jtreg/compiler/c2/irTests/TestPadding.java [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+        TestFramework.runWithFlags("-XX:+IntelJccErratumMitigation");
+    }
+
+    @Run(test = "test")
+    public static void test_runner() {
+        tpf = new TestPadding();
+        for (int i = 0; i < 11000; i++) {
+            test(i);
+            tpf.b1++; // to take both branches in test()
+        }
```
Comment (chhagedorn) [inline test/hotspot/jtreg/compiler/c2/irTests/TestPadding.java:50]: `test_runner()` will be invoked 2000 times (default warm-up) before compilation of `test()`. So, this loop will run 2000 * 11000 times. Do you need that many iterations or can the loop be removed such that we only have 2000 warm-up iterations? I.e. something like:
```
    @Run(test = "test")
    public static void test_runner() {
        tpf = new TestPadding();
        test(42);
        tpf.b1++; // to take both branches in test()

    }
```
If you need more iterations, you could still specify `@Warmup(12345)` at `test_runner()` to get more profiling in before compilation of `test()`.

**Review by chhagedorn — APPROVED** [review]
Otherwise, the fix looks good. Thanks for adding an IR test!

**Review by TobiHartmann — APPROVED** [review]
Looks good to me.

**Review by shipilev — APPROVED** [review]
Looks fine to me, thanks.

**Comment by vnkozlov** [comment]
Thank you, Alexey and Tobias, for reviews.

**Comment by vnkozlov** [comment]
`StressStackOverflow.java` test failure in GHA on 32-bit x86 I saw in other new PRs too.

File: test/hotspot/jtreg/compiler/c2/irTests/TestPadding.java (hunk 1 above)
Comment (chhagedorn) [reply test/hotspot/jtreg/compiler/c2/irTests/TestPadding.java]: That looks good to me, thanks for the update!

**Review by chhagedorn — APPROVED** [review]

**Pull request closed without merging**